src/Core/Form/ChoiceProvider/LanguageByIdChoiceProvider.php

```diff
@@ -40,11 +40,20 @@ final class LanguageByIdChoiceProvider implements FormChoiceProviderInterface
     private $languageDataProvider;
 
     /**
+     * @var int
+     */
+    private $shopId;
+
+    /**
+     * LanguageByIdChoiceProvider constructor.
+     *
      * @param LanguageDataProvider $languageDataProvider
+     * @param int $shopId
      */
-    public function __construct(LanguageDataProvider $languageDataProvider)
+    public function __construct(LanguageDataProvider $languageDataProvider, int $shopId)
```

Comment:
It's a BC break! Not only does it change the constructor of an existing class It completely changes its behavior because it will only return the language for a specific shop

In your use case you want to filter for one shop but it doesn't mean it's always used for a single shop

I see two alternatives you can change the interface `ConfigurableFormChoiceProviderInterface` so you can use an `$options` containing the shop_id (optional)
Or create a new provider altogether

Reply:
I'm more in favor of using the ConfigurableFormChoiceProviderInterface to be consistent with other classes :)

Reply:
I'll do that ! I totally didn't think about the bc break...